dvc/api.py

```diff
@@ -69,3 +95,91 @@ def _make_repo(repo_url, rev=None):
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
             yield repo
+
+
+def summon(name, fname="dvcsummon.yaml", args=None, repo=None, rev=None):
+    # TODO: Write a meaningful docstring about `summon`
+    with _make_repo(repo, rev=rev) as _repo:
+
+        def pull_dependencies(deps):
+            if not deps:
+                return
+
+            outs = [_repo.find_out_by_relpath(dep) for dep in deps]
+
+            with _repo.state:
+                for out in outs:
+                    _repo.cloud.pull(out.get_used_cache())
+                    out.checkout()
+
+        path = os.path.join(_repo.root_dir, fname)
+        obj = _get_object_from_summoners_file(name, path)
+        info = obj["summon"]
+
+        pull_dependencies(info.get("deps"))
+
+        _args = copy.deepcopy(info.get("args", {}))
+        _args.update(args or {})
+
+        return _invoke_method(info["call"], _args, path=_repo.root_dir)
+
+
+def _get_object_from_summoners_file(name, path):
+    """
+    Given a summonable object's name, search for it on the given file
+    and bring it to life.
+    """
+    try:
+        with open(path, "r") as fobj:
+            content = summon_schema(ruamel.yaml.safe_load(fobj.read()))
+            objects = content["objects"]
+            return next(x for x in objects if x["name"] == name)
+    except FileMissingError:
+        # XXX: Prints ugly absolute path.
+        raise SummonError("No such YAML file with path: '{}'".format(path))
+    except ruamel.yaml.YAMLError as exc:
+        raise SummonError("Failed to parse YAML correctly", exc)
+    except Invalid as exc:
+        # XXX: Doesn't point out the error.
+        raise SummonError("YAML file dosen't match with the schema", exc)
+    except StopIteration:
+        raise SummonError("No such object with name '{}'".format(name))
```

Comment:
Again, need to add summon_file and repo. We might actually add it outside where this info is available. This will fix all of the above and wouldn't need those two be passed here.